src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -1,42 +1,62 @@
 import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
-import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
 import { OrderActionCreator, OrderRequestBody } from '../../../order';
 import { OrderFinalizationNotRequiredError } from '../../../order/errors';
 import { PaymentArgumentInvalidError } from '../../errors';
 import PaymentActionCreator from '../../payment-action-creator';
 import { PaymentRequestOptions } from '../../payment-request-options';
 import PaymentStrategy from '../payment-strategy';
 
+import { PaypalCommercePaymentProcessor, PaypalCommerceRequestSender } from './index';
+
 export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
 
     constructor(
         private _store: CheckoutStore,
         private _orderActionCreator: OrderActionCreator,
-        private _paymentActionCreator: PaymentActionCreator
+        private _paymentActionCreator: PaymentActionCreator,
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender,
+        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor
     ) {}
 
-    initialize(): Promise<InternalCheckoutSelectors> {
+    async initialize(): Promise<InternalCheckoutSelectors> {
         return Promise.resolve(this._store.getState());
     }
 
     async execute(payload: OrderRequestBody, options: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
         const { payment, ...order } = payload;
-        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethod(options.methodId);
+        const state = this._store.getState();
+        const paymentMethod = state.paymentMethods.getPaymentMethod(options.methodId);
 
         if (!payment) {
             throw new PaymentArgumentInvalidError(['payment']);
         }
 
-        if (!paymentMethod || !paymentMethod.initializationData.orderId) {
-            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        if (!paymentMethod) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        let { orderId } = paymentMethod.initializationData;
+
+        if (!orderId) {
+            const cart = state.cart.getCart();
```

Comment:
Perhaps you can add a `getCartOrThrow` method to `CartSelector` to simplify this.

Reply:
Great! I'll add it. Thank you